agent/api/task/task.go

```diff
@@ -202,55 +206,53 @@ func (task *Task) PostUnmarshalTask(cfg *config.Config,
 			return apierrors.NewResourceInitError(task.Arn, err)
 		}
 	}
-	task.initializeEmptyVolumes()
+	task.initializeLocalVolumes(dockerClient)
 	task.initializeCredentialsEndpoint(credentialsManager)
 	task.addNetworkResourceProvisioningDependency(cfg)
 	return nil
 }
 
-func (task *Task) initializeEmptyVolumes() {
-	requiredEmptyVolumes := []string{}
+func (task *Task) initializeLocalVolumes(dockerClient dockerapi.DockerClient) {
+	requiredLocalVolumes := []string{}
+	reInvalidChars := regexp.MustCompile("[^A-Za-z0-9-]+")
 	for _, container := range task.Containers {
 		for _, mountPoint := range container.MountPoints {
 			vol, ok := task.HostVolumeByName(mountPoint.SourceVolume)
 			if !ok {
 				continue
 			}
-			if _, ok := vol.(*EmptyHostVolume); ok {
-				container.BuildContainerDependency(emptyHostVolumeName, apicontainer.ContainerRunning, apicontainer.ContainerCreated)
-				requiredEmptyVolumes = append(requiredEmptyVolumes, mountPoint.SourceVolume)
+			if localVolume, ok := vol.(*taskresourcevolume.LocalVolume); ok {
+				name := reInvalidChars.ReplaceAllString(mountPoint.SourceVolume, "")
+				localVolume.HostPath = task.taskScopedVolumeName(name)
+				container.BuildResourceDependency(mountPoint.SourceVolume,
+					taskresource.ResourceStatus(taskresourcevolume.VolumeCreated),
+					apicontainer.ContainerPulled)
+				requiredLocalVolumes = append(requiredLocalVolumes, mountPoint.SourceVolume)
 			}
 		}
 	}
 
-	if len(requiredEmptyVolumes) == 0 {
-		// No need to create the auxiliary 'empty-volumes' container
+	if len(requiredLocalVolumes) == 0 {
+		// No need to create the auxiliary local driver volumes
 		return
 	}
 
-	// If we have required empty volumes, add an 'internal' container that handles all
-	// of them
-	_, ok := task.ContainerByName(emptyHostVolumeName)
-	if !ok {
-		mountPoints := make([]apicontainer.MountPoint, len(requiredEmptyVolumes))
-		for i, volume := range requiredEmptyVolumes {
-			// BUG(samuelkarp) On Windows, volumes with names that differ only by case will collide
-			containerPath := getCanonicalPath(emptyvolume.ContainerPathPrefix + volume)
-			mountPoints[i] = apicontainer.MountPoint{SourceVolume: volume, ContainerPath: containerPath}
-		}
-		sourceContainer := &apicontainer.Container{
-			Name:                emptyHostVolumeName,
-			Image:               emptyvolume.Image + ":" + emptyvolume.Tag,
-			Command:             []string{emptyvolume.Command}, // Command required, but this only gets created so N/A
-			MountPoints:         mountPoints,
-			Essential:           false,
-			Type:                apicontainer.ContainerEmptyHostVolume,
-			DesiredStatusUnsafe: apicontainer.ContainerRunning,
-		}
-		task.Containers = append(task.Containers, sourceContainer)
+	// if we have required local volumes, create one with default local drive
+	for _, volumeName := range requiredLocalVolumes {
+		vol, _ := task.HostVolumeByName(volumeName)
```

Comment:
Im not sure how this code interacts with the Windows disk manager, so I dont have a good solution, but I am concerned that we arent using a unique identifier here.

What are we doing to ensure we are choosing the right volume and arent conflicting spaces between tasks?